homework-assignments-1/task_4.ts

```diff
@@ -0,0 +1,170 @@
+class Product {
+    name:string;
+    price:number;
+    stock:number;
+    constructor(name:string, price:number, stock:number) {
+        this.name = name;
+        this.price = price;
+        this.stock = stock;
+    }
+
+    addStock(quantity:number):void{
+        this.stock += quantity;
+        console.log(`${quantity} items added. New stock: ${this.stock}`);
+    };
+    removeStock(quantity:number):void{
+        this.stock -= quantity;
+        console.log(`${quantity} items removed. New stock: ${this.stock}`);
+    }
+}
+
+class Customer {
+    name: string;
+    email:string;
+    cart: Product[]
+    address:string;
+
+    constructor(name:string, email:string, address:string) {
+        this.name = name;
+        this.email = email;
+        this.cart = [];
+        this.address = address;
+    }
+    addToCart(product:Product):void{
+        this.cart.push(product);
```

Comment:
Don't push product to cart directly. Check if there are any stocks in market if (product.getStock() > 0)